modules/ting/ting.field.inc

```diff
@@ -363,8 +363,18 @@ function ting_field_formatter_view($entity_type, $entity, $field, $instance, $la
           $ting_type = $entity->getType();
           if (count($collection->types) > 1) {
             $ting_type = t('Material collection');
-            // Replace title with the more generic collection title.
-            $title = $collection->title;
+          }
+          $collection_entity_title = NULL;
+          foreach ($collection->entities as $collection_entity) {
+            if ($collection_entity_title == NULL) {
+              $collection_entity_title = $collection_entity->title;
+            }
+            elseif ($collection_entity_title != $collection_entity->title) {
+              // Replace title with the more generic collection title.
+              $title = $collection->title;
+              $ting_type = t('Material collection');
+              break;
+            }
```

Comment:
I have tried reading through this loop + conditionals multiple times and I am still having trouble wrapping my head around the code here.

After having read the issue again then how about changing it to something like this:

```php
$collection_titles = array_map(function($collection_entity) {
  return $collection_entity->title;
});
// If a collection has items of the same type but with different titles then use the generic collection title
if (count(array_unique($collection_titles) > 1) {
    $title = $collection->title;
    $ting_type = t('Material collection');
}
```

**Note: The code is completely untested!**